### PR TITLE
V8: ClipboardService + implementation for nested content

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Developed a ClipboardService and implementing it into Nested Content.

Based on the concept from https://github.com/umbraco/Umbraco-CMS/pull/2963/files by @PeterBridgerZone.

The PR consists of a CopyService, which can be used across different data-types.
And to prove the work I have implemented it into Nested Content.

This PR will also change the look of Nested Content sligthly to now look like this:

image

To test this PR, you should copy some content from a nested content property, paste it into the same or another one. You can also copy during splitscreen mode.

Notice the copy-service has no limits to how much data it can contain, so you can copy from several places and it will be kept in the clipboard, to be pasted later.

In this way you can copy several items from a ContentNode, navigate to another one to paste the items that you want.

This change also includes a few visual updates to ItemPicker for NestedContent and the NestedContent it self has also gotten a few updates those including:

- You can now sort items directly by dragging the header of a item.
- The edit button is gone, since you edit by clicking the header of a item.
- A copy button has been added.
- The ItemPicker now has the ability to show items that are possible to paste from the clipboard.
- The ItemPicker now has the ability to remove paste-able items from the clipboard.

I think thats it. :-)

### CODE REVIEW

**Comment by bjarnef** [comment]
@nielslyngsoe maybe `umbCopyService` would be a better storage key name than `uCopyService`?

**Comment by warrenbuckley** [comment]
All tested and works - merged in

File: src/Umbraco.Web.UI.Client/src/less/components/card.less [hunk 1]
```diff
@@ -84,63 +85,73 @@
...
     max-width:33.33%;
 }
 
 .umb-card-grid .umb-card-grid-item {
+    position: relative;
 	display: block;
 	width: 100%;
-	height: 100%;
+	//height: 100%;
+    padding-top: 100%;
```
Comment (bjarnef) [inline src/Umbraco.Web.UI.Client/src/less/components/card.less]: @nielslyngsoe it seems this `padding-top: 100%` afffect field types in Umbraco Forms (which use the `.umb-card` styles.

image

Comment (nielslyngsoe) [reply src/Umbraco.Web.UI.Client/src/less/components/card.less]: Hi @bjarnef thanks, The issue is also reported on the Forms Issue Tracker. https://github.com/umbraco/Umbraco.Forms.Issues/issues/217

I hope we will get it fixed soon — We shouldn't be using css-classes directly between projects, only components, but its not always the case.

**Pull request merged**